File: my_spaceship/ex00/my_spaceship.js
function my_spaceship(str) {
    let xAxis = 0;
    let yAxis = 0;
    let dir = 'up';

    for (i = 0; i < str.length; i++) {
        if (str[i] == 'A') {
            if (dir == 'up') {
                yAxis--;
                dir = 'up';
                }
            else if (dir == 'down') {
                yAxis++
                dir = 'down';
            }
            else if (dir == 'right') {
                xAxis++;
                dir = 'right';
            } else if (dir == 'left') {
                xAxis--;
                dir = 'left';
            }
        }
        else if (str[i] == 'R') {
               if (dir == 'right') {
                dir = 'down';
               }
               else if (dir == 'down') {
                dir = 'left';
            } else if (dir == 'left') {
                dir = 'up';
            } else if (dir == 'up') {
                dir = 'right';
            }
        }
        else if (str[i] == 'L') {
            if (dir == 'left') {
                dir = 'down';
               }
             else if (dir == 'down') {
                dir = 'right';
            } else if (dir == 'right') {
                dir = 'up';
            } else if (dir == 'up') {
                dir = 'left';
            }
        }
        }

           // res = "{x: "+x+ " , y : " +y+ " , direction : '"+dir+ "'}";
            return  `{x: ${xAxis}, y: ${yAxis}, direction: '${dir}'}`
            
    
    }
    
     

File: quest04/ex05/my_spaceship.c
#include <stdio.h>
#include <string.h>
#include <stdlib.h>

char* my_spaceship(char* param_1)
{
int x,y = 0;
int current_direction = 0;
char* directions[4] = {"up", "right", "down", "left"};
//               0       1       2        3 
for (int i=0;param_1[i] != '\0';i++) {
    if (param_1[i] == 'R') {
       current_direction += 1;
    } 
    else if (param_1[i] == 'L') {
       current_direction -= 1;
       if (current_direction == -1) {
       current_direction = 3;
       }
    } 
    else {
        if (current_direction == 0) {
        y--;
        }
        else if (current_direction == 1) {
        x++;
        }
        else if (current_direction == 2) {
        y++;
        } 
        else {
        x--;
        }
    }
    
    }
char* res = malloc(50);

snprintf(res, 50, "{x: %d, y: %d, direction: '%s'}", x-1, y, directions[current_direction]);

return res;
}

/*int main() {
char* str = "RAALALL";
my_spaceship(str);
}*/